sdk/kiosk/src/client/kiosk-client.ts

```diff
@@ -0,0 +1,485 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiClient } from '@mysten/sui.js/src/client';
+import { fetchKiosk, getOwnedKiosks } from '../query/kiosk';
+import {
+	type FetchKioskOptions,
+	KioskData,
+	KioskOwnerCap,
+	ObjectArgument,
+	OwnedKiosks,
+	Network,
+	type KioskClientOptions,
+} from '../types';
+import { TransactionArgument, TransactionBlock } from '@mysten/sui.js/transactions';
+import * as kioskTx from '../tx/kiosk';
+import { TransferPolicyRule, personalKioskAddress, rules } from '../constants';
+import { queryTransferPolicy } from '../query/transfer-policy';
+import { convertToPersonalTx } from '../tx/personal-kiosk';
+import { confirmRequest } from '../tx/transfer-policy';
+import { objArg } from '../utils';
+
+export type PurchaseOptions = {
+	extraArgs?: Record<string, any>;
+};
+
+/**
+ * A Client that allows you to interact with kiosk.
+ * Offers utilities to query kiosk, craft transactions to edit your own kiosk,
+ * purchase, manage transfer policies, create new kiosks etc.
+ */
+export class KioskClient {
```

Comment:
I feel like this client is doing a little too much currently. It seems like it's trying to be responsible for fetching data, and also transaction construction, but those are really two separate groups of functionality that I imagine folks will use in different contexts. e.g. explorer will only care about fetching data, but it drags all of the transaction construction along for the ride, which feels like a big miss here.